### PR TITLE
ENH: Implement str.extract for ArrowDtype

### LINKED ISSUE #56268: BUG: `str.extract` Method Not Implemented for `pd.ArrowDtype(pa.string())`
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [ ] I have confirmed this bug exists on the [main branch](https://pandas.pydata.org/docs/dev/getting_started/install.html#installing-the-development-version-of-pandas) of pandas.

### Reproducible Example

```python
import pandas as pd
import io

data ='''make,city08
Alfa Romeo,19
Ferrari,9
Dodge,23
Dodge,10
Subaru,17'''

df = pd.read_csv(io.StringIO(data), dtype_backend='pyarrow')
df.make.str.extract(r'([^a-z A-Z])')
```

### Issue Description

The `str.extract` method raises a `NotImplementedError` when used on a series with the `pd.ArrowDtype(pa.string())` data type.

### Expected Behavior

Execute like it does with Pandas 1 strings.

### Installed Versions

<details>

the `str.extract` method raises a `NotImplementedError` when used on a series with the `pd.ArrowDtype(pa.string())` data type.

</details>

### PR DESCRIPTION
**Pull request by mroeschke**
- [x] closes #56268 (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/tests/extension/test_arrow.py [hunk 1]
```diff
@@ -2150,14 +2150,39 @@ def test_str_rsplit():
...
+    ser = pd.Series(["a1", "b2", "c3"], dtype=ArrowDtype(pa.string()))
+    result = ser.str.extract(r"[ab](?P<digit>\d)", expand=True)
+    expected = pd.DataFrame(
+        {
+            "digit": ArrowExtensionArray(pa.array(["1", "2", None])),
+        }
+    )
+    tm.assert_frame_equal(result, expected)
+
+    result = ser.str.extract(r"[ab](?P<digit>\d)", expand=False)
```
Comment (WillAyd) [inline pandas/tests/extension/test_arrow.py:2181]: Can we add coverage for when there are multiple groups and `expand=False`?

Comment (mroeschke) [reply pandas/tests/extension/test_arrow.py:2182]: Sure, I parametrized the above test with `expand=`

Comment (WillAyd) [reply pandas/tests/extension/test_arrow.py:2182]: Shouldn't the `expand=False` case only produce a series?

Comment (mroeschke) [reply pandas/tests/extension/test_arrow.py:2182]: Not necessarily:

https://pandas.pydata.org/docs/reference/api/pandas.Series.str.extract.html

> If True, return DataFrame with one column per capture group. If False, return a Series/Index if there is one capture group or DataFrame if there are multiple capture groups.

Comment (WillAyd) [reply pandas/tests/extension/test_arrow.py:2182]: Ah OK - sorry I misremembered how that works. I was somewhat expecting `expand=False` here to just return the struct array

Comment (WillAyd) [reply pandas/tests/extension/test_arrow.py:2182]: FWIW I'm OK with this now too as it is definitely consistent with our current API. There may be a future where we actuallly want it to return a pyarrow struct but can always come back and do that later

Comment (mroeschke) [reply pandas/tests/extension/test_arrow.py:2182]: Yeah agreed. For arrow types, I think in the future `expand=False` should return a struct typed Series

**Review by phofl — APPROVED** [review]
thx

**Pull request merged**